opennlp-dl/src/main/java/opennlp/dl/doccat/DocumentCategorizerDL.java

```diff
@@ -0,0 +1,166 @@
+     * @param categories The categories.
+     */
+    public DocumentCategorizerDL(File model, File vocab, Map<Integer, String> categories) {
+
+        this.model = model;
+        this.vocab = vocab;
+        this.categories = categories;
+
+    }
+
+    @Override
+    public double[] categorize(String[] strings) {
+
+        try {
+
+            final DocumentCategorizerInference inference = new DocumentCategorizerInference(model, vocab);
+
+            final double[][] vectors = inference.infer(strings[0]);
+            final double[] results = inference.softmax(vectors[0]);
+
+            return results;
+
+        } catch (Exception ex) {
+            System.err.println("Unload to perform document classification inference: " + ex.getMessage());
+        }
+
+        return new double[]{};
+
+    }
+
+    @Override
+    public double[] categorize(String[] strings, Map<String, Object> map) {
+        return categorize(strings);
+    }
+
+    @Override
+    public String getBestCategory(double[] doubles) {
+        return categories.get(maxIndex(doubles));
+    }
+
+    @Override
+    public int getIndex(String s) {
+        return getKey(s);
+    }
+
+    @Override
+    public String getCategory(int i) {
+        return categories.get(i);
+    }
+
+    @Override
+    public int getNumberOfCategories() {
+       return categories.size();
+    }
+
+    @Override
+    public String getAllResults(double[] doubles) {
+        return null;
+    }
+
+    @Override
+    public Map<String, Double> scoreMap(String[] strings) {
+
+        final double[] scores = categorize(strings);
+
+        final Map<String, Double> scoreMap = new HashMap<>();
+
+        for(int x : categories.keySet()) {
+            scoreMap.put(categories.get(x), scores[x]);
+        }
+
+        return scoreMap;
+
+    }
+
+    @Override
+    public SortedMap<Double, Set<String>> sortedScoreMap(String[] strings) {
+
+        final double[] scores = categorize(strings);
+
+        final SortedMap<Double, Set<String>> scoreMap = new TreeMap<>();
+
+        for(int x : categories.keySet()) {
+
+            if(scoreMap.get(scores[x]) == null) {
+                scoreMap.put(scores[x], new HashSet<>());
+            }
+
+            scoreMap.get(scores[x]).add(categories.get(x));
+
+        }
+
+        return scoreMap;
+
+    }
+
+    private int getKey(String value) {
+
+        for (Map.Entry<Integer, String> entry : categories.entrySet()) {
+
+            if (entry.getValue().equals(value)) {
+                return entry.getKey();
+            }
+
+        }
+
+        // The String wasn't found as a value in the map.
+        return -1;
+
+    }
+
+    private int maxIndex(double[] arr) {
+
+        double max = Double.NEGATIVE_INFINITY;
+        int index = -1;
+
+        for(int x = 0; x < arr.length; x++) {
+            if(arr[x] > max) {
+                index = x;
+                max = arr[x];
+            }
+        }
+
+        return index;
```

Comment:
Ditto about calculating max here. Maybe this could even be a common static method somewhere, and re-use it instead of maintaining the duplicated code.